src/components/messenger/list/conversation-list-panel/index.tsx

```diff
@@ -116,6 +117,7 @@ export class ConversationListPanel extends React.Component<Properties, State> {
 
   openExistingConversation = (id: string) => {
     this.props.onConversationClick(id);
+    this.props.markConversationAsRead({ channelId: id });
```

Comment:
Seems like maybe this logic should be in the sagas. Opening the conversation should already be calling out to redux/saga to set the active conversation id. I think that can be a saga that also marks the conversation as read rather than relying on UI to remember to do both of these things every time.

Reply:
The reason i didn't do it that way because we're setting the activeConversationId in a lot of other places as well (for eg. creating a new conversation, onNewConversationCreated hook etc). On the UI side it was only 2 places where i had to place this (one here, and other when we set active channel ID). Not sure which decision is better here

Reply:
Why wouldn't we want to mark messages as read on those other cases too?

Reply:
I think one issue here is that the name we've used isn't truly representative of what we're doing any more. `setActiveConversationId` isn't truly what we're doing when you look at it from a system perspective. It's too technical of a name. Really what we want to do is `openConversation` or something along those lines. When you look at it that way, it makes sense that what we're trying to say is "any time we open a conversation we should mark it as read".

Reply:
> Why wouldn't we want to mark messages as read on those other cases too?

I thought when we're creating a "new" conversation we wouldn't have any messages to be "marked as read". 😅

Reply:
> I think one issue here is that the name we've used isn't truly representative of what we're doing any more. setActiveConversationId isn't truly what we're doing when you look at it from a system perspective. It's too technical of a name. Really what we want to do is openConversation or something along those lines. When you look at it that way, it makes sense that what we're trying to say is "any time we open a conversation we should mark it as read".

That makes sense i think. Let me do the updates here, and create a follow up PR for the "renaming".

Reply:
> I thought when we're creating a "new" conversation we wouldn't have any messages to be "marked as read"

Yeah, that's probably the 1 case where we don't need to (and might fail if we don't have a proper channel id yet) but that would be the edge case, right? There could be so many ways to open a conversation. For example, the in app notifications might become clickable. In which case, we'd be opening a conversation from there and would want this behavior.

My point being, the majority case is that we'd want to do this...and we don't want to have to remember to do that everywhere.

Reply:
did the updates.